src/wallet-metrics.ts

```diff
@@ -0,0 +1,388 @@
+          alias: `low-master-sponsor-balance-${wallet.chainName}`,
+          priority: 'P3',
+        },
+        walletConfig.opsGenieConfig
+      );
+    }
+    await opsGenie.closeOpsGenieAlertWithAlias(
+      `low-master-sponsor-balance-${wallet.chainName}`,
+      walletConfig.opsGenieConfig
+    );
+
+    const gasOptions = await getGas(globalSponsor);
+
+    if (!process.env.WALLET_ENABLE_SEND_FUNDS) {
+      await opsGenie.sendToOpsGenieLowLevel(
+        {
+          message: `(would have) Just topped up ${wallet.address} on ${wallet.chainName}`,
+          alias: `freshly-topped-up-${wallet.address}-${wallet.chainName}`,
+          description: [
+            `DID NOT ACTUALLY SEND FUNDS! WALLET_ENABLE_SEND_FUNDS is not set`,
+            `Type of wallet: ${wallet.walletType}`,
+            `Address: ${evm.resolveExplorerUrlByName(walletConfig.explorerUrls, wallet.chainName)}address/${
+              wallet.address
+            } )`,
+            `Transaction: ${evm.resolveExplorerUrlByName(
+              walletConfig.explorerUrls,
+              wallet.chainName
+            )}tx/not-applicable`,
+          ].join('\n'),
+          priority: 'P5',
+        },
+        walletConfig.opsGenieConfig
+      );
+
+      await opsGenie.closeOpsGenieAlertWithAlias(
+        `error-while-topping-up-wallet-${wallet.address}-${wallet.chainName}`,
+        walletConfig.opsGenieConfig
+      );
+
+      return;
+    }
+
+    const receipt = await globalSponsor.sponsor.sendTransaction({
+      to: wallet.address,
+      value: parseEther(globalSponsor.topUpAmount),
+      gasPrice: gasOptions?.gasPrice,
+    });
+    await receipt.wait(1);
+
+    await opsGenie.sendToOpsGenieLowLevel(
+      {
+        message: `Just topped up ${wallet.address} on ${wallet.chainName}`,
+        alias: `freshly-topped-up-${wallet.address}-${wallet.chainName}`,
+        description: [
+          `Type of wallet: ${wallet.walletType}`,
+          `Address: ${evm.resolveExplorerUrlByName(walletConfig.explorerUrls, wallet.chainName)}address/${
+            wallet.address
+          } )`,
+          `Transaction: ${evm.resolveExplorerUrlByName(walletConfig.explorerUrls, wallet.chainName)}tx/${
+            receipt?.hash ?? 'WALLET_ENABLE_SEND_FUNDS disabled'
+          }`,
+        ].join('\n'),
+        priority: 'P5',
+      },
+      walletConfig.opsGenieConfig
+    );
+
+    await opsGenie.closeOpsGenieAlertWithAlias(
+      `error-while-topping-up-wallet-${wallet.address}-${wallet.chainName}`,
+      walletConfig.opsGenieConfig
+    );
+  } catch (e) {
+    await opsGenie.sendToOpsGenieLowLevel(
+      {
+        message: 'An error occurred while trying to up up a wallet',
+        alias: `error-while-topping-up-wallet-${wallet.address}-${wallet.chainName}`,
+        priority: 'P1',
+        description: `Error: ${e}
+      Stack Trace: ${(e as Error)?.stack}`,
+      },
+      walletConfig.opsGenieConfig
+    );
+
+    return;
+  }
+};
+
+/**
```

Comment:
This section is probably a great opportunity to use your work from Airnode @ gas oracle (for a later task).

Reply:
Task for this [here](https://api3dao.atlassian.net/browse/BEC-641)